src/core.js

```diff
@@ -1,6 +1,44 @@
 'use strict';
+import {Model} from './model';
+import {serialize, deserialize} from './serializers/xml';
+import {persist} from './adapters/mediawiki';
+
+export {Core as InfoboxTemplateBuilder};
+
+
+class Core extends Model {
+
+	constructor(params = {}) {
+
+		super();
+
+		const {from} = params;
+
+		/*
+		 * If builder is instantiated with a serialized document, we will deconstruct it
+		 * into our internal representation, and populate the builder with those values
+		 */
+		if (from) {
+
+			[this.data, this.theme] = deserialize(from);
```

Comment:
Looks like we don't get to destructure objects like `{foo: 'foo', bar: 'bar'}` into arrays, only arrays, like `['foo', 'bar']`. Error I'm seeing is `Uncaught (in promise) Error: Invalid attempt to destructure non-iterable instance`.